lib/HTFeed/JobMetrics.pm

```diff
@@ -0,0 +1,320 @@
+package HTFeed::JobMetrics;
+
+use strict;
+use warnings;
+
+use Log::Log4perl qw(get_logger);
+use Prometheus::Tiny::Shared;
+
+my $USAGE = q^
+# This module is responsible for measuring job performance in different
+# stages. It knows what kinds of things we want to measure, and passes
+# each measurement to Prometheus, which writes to a Hash::SharedMem.
+
+# Note that metric names are case sensitive and often ucfirst.
+# To see metric names, see --list.
+
+### Usage:
+#
+### show the current metrics:
+# $ perl JobMetrics.pm --pretty
+#
+### show where on the filesystem metrics are stored:
+# $ perl JobMetrics.pm --loc
+#
+### just see the list of metrics
+# $ perl JobMetrics.pm --list
+#
+### irrevocably clear current metrics:
+# $ perl JobMetrics.pm --clear
+#
+### add value to a counter:
+# $ perl JobMetrics.pm --add <metric> <value>
+#
+### add value to histogram metric:
+# $ perl JobMetrics.pm --observe <metric> <value>
+#
+### increment counter metric:
+# $ perl JobMetrics.pm --inc <metric>
+#
+### or:
+# use HTFeed::JobMetrics;
+# my $jm = HTFeed::JobMetrics->new();
+# $jm->inc("Handle_items");         # for counters    (++)
+# $jm->add("Collate_bytes", 2460) ;     # for counters    (+= x)
+# $jm->observe("Download_ms", 220); # for historgrams (+= x)
+# print $jm->pretty;
+### list available metrics
+# $jm->list_metrics
+^;
+
+# If called from commandline:
+if (!caller) {
+    my $jm = HTFeed::JobMetrics->new();
+
+    if (!@ARGV || grep { /--(help|usage)/ } @ARGV) {
+	print $USAGE;
+	exit;
+    }
+
+    while (my $arg = shift @ARGV) {
+	if ($arg =~ /^--pretty$/) {
```

Comment:
if we retain this functionality in the final version I would want to see it use `Getopt::Long`

Reply:
I'll look at that. 

It seemed excessive to use Getopt at first, then the CLI code grew into something that definitely needs to be managed... so yeah, agree.

Reply:
Now using `GetOpt::Long`.